### DISCUSSION TITLE
Special \charGenerate function to be used inside \evalWhole?

### DISCUSSION THREAD

**Discussion by user202729** [lvjr/functional#11] [Ideas]
I think it would be convenient. Could be used like this

```
\evalWhole { 123 \charGenerate{`A}{1} 456 \charGenerate{`A}{2} }
```

For generating tokens of cat 1 and 2, it requires some "special support" from `\evalWhole` (and would not make sense outside of it), but putting the raw `\char_generate:nn` into the resulting token list to be x-expanded should be sufficient. Other catcodes can be used both inside and outside.

**Comment by lvjr** [comment]
Sorry, I am not able to understand very well what the code is doing or trying to do? Generating two characters `A` with catcodes `1` and `2`, then ... ? Do you mean the code will throw an error?

**Reply by user202729** [reply]
I want to be able to execute some token lists that contains special-charcode tokens. For example the desired output of

```
\evalWhole { \tl_analysis_show:n { \charGenerate {`A}{1} \charGenerate {`A}{2} } }
```

should be that the token list consist of 2 tokens, one begin-group `A` and one end-group `A`.

**Reply by lvjr** [reply]
Then you may need to write like this
```
 \tlAnalysisShow { \evalWhole { 123 \charGenerate{`A}{1} 456 \charGenerate{`A}{2} } }
```
Need to define `\tlAnalysisShow` and `\charGenerate` first.

**Reply by user202729** [reply]
No... my point is that `charGenerate` with catcode 1 or 2 cannot be implemented in the current framework (as far as I can see), thus need special casing.

Actually... thinking about it it's possible to implement `\charGenerateGroup{`A}{456}{`A}` which evaluates to `(A with catcode 1) 456 (A with catcode 2)` as well. So... can "close" this I guess.

**Reply by lvjr** [reply]
We know `functional` is quite different from `expl3`, so we can expect many functions in `expl3` can not be transfered to `functional`.
